File: backend/mallapi/src/main/java/org/nareun130/mallapi/formatter/LocalDateFormatter.java
package org.nareun130.mallapi.formatter;

import java.text.ParseException;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Locale;

import org.springframework.format.Formatter;

//* 날짜 시간을 다루기 위한 포매터
public class LocalDateFormatter implements Formatter<LocalDate>{

    @Override
    public LocalDate parse(String text, Locale locale) {
        
        return LocalDate.parse(text, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
    }

    @Override
    public String print(LocalDate object, Locale locale) {

        return DateTimeFormatter.ofPattern("yyyy-MM-dd").format(object);
    }

    
}

File: backend/mallapi-kotlin/src/main/kotlin/com/nareun/mallapikotlin/controller/formatter/LocalDateFormatter.kt
package com.nareun.mallapikotlin.controller.formatter

import org.springframework.context.annotation.Configuration
import org.springframework.format.Formatter
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.util.Locale

// LocalDate와 LocalDateTime 포맷을 통합시켜주는 formatter
@Configuration
class LocalDateFormatter : Formatter<LocalDate> {


    override fun parse(text: String, locale: Locale): LocalDate {
        return LocalDate.parse(text, DateTimeFormatter.ofPattern("yyyy-MM-dd"))
    }

    override fun print(localDate: LocalDate, locale: Locale): String {
        return DateTimeFormatter.ofPattern("yyyy-MM-dd").format(localDate)
    }

}